### PR TITLE
[SYCL] Add tests for SYCL2020 sub_groups features

### PR DESCRIPTION
**Pull request by alexbatashev**
Link to feature: https://github.com/intel/llvm/pull/3765

### CODE REVIEW

File: SYCL/SubGroup/barrier.cpp [hunk 1]
```diff
@@ -39,7 +51,8 @@ template <typename T> void check(queue &Queue, size_t G = 240, size_t L = 60) {
         for (size_t i = 0; i <= lid; i++) {
           res += addacc[SGoff + i];
         }
-        SG.barrier(access::fence_space::global_space);
+        // SG.barrier(access::fence_space::global_space);
```
Comment (Pennycook) [inline SYCL/SubGroup/barrier.cpp]: Did you mean to delete this comment?

File: SYCL/SubGroup/sub_groups_sycl2020.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  sycl::host_accessor Acc{Buf, sycl::read_only};
+
+  sycl::kernel_id TestKernelID = sycl::get_kernel_id<TestKernel>();
+  sycl::kernel Kernel = KernelBundle.get_kernel(TestKernelID);
+
+  const size_t MaxNumSubgroups =
+      Kernel.get_info<sycl::info::kernel_device_specific::max_num_sub_groups>(
+          Q.get_device());
+  const size_t SubgroupSize = 32 / MaxNumSubgroups;
```
Comment (Pennycook) [inline SYCL/SubGroup/sub_groups_sycl2020.cpp]: Any reason not to use `info::kernel_device_specific::max_sub_group_size` instead?

**Review by Pennycook — DISMISSED** [review]

File: SYCL/SubGroup/sub_groups_sycl2020.cpp (hunk 2 above)
Comment (alexbatashev) [reply SYCL/SubGroup/sub_groups_sycl2020.cpp]: That's actually a nice piece of advice. `kernel_device_specific::max_num_sub_groups` returns maximum number of subroups, not the one that was actually run. I updated implementation to use the correct values.

**Review by Pennycook — DISMISSED** [review]

**Comment by vladimirlaz** [comment]
@alexbatashev, the newly added test is failing on CUDA. Could you please fix it?

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
@alexbatashev, the newly added test is failing on CUDA. Could you please fix it?

**Comment by alexbatashev** [comment]
> @alexbatashev, the newly added test is failing on CUDA. Could you please fix it?

I disabled test on CUDA, because sycl tools seemingly do not generate enough information for the runtime to function correctly. I'll create a tracker for that.

As for windows regression, it does not seem to be related to my patch.

**Comment by vladimirlaz** [comment]
> > @alexbatashev, the newly added test is failing on CUDA. Could you please fix it?
> 
> I disabled test on CUDA, because sycl tools seemingly do not generate enough information for the runtime to function correctly. I'll create a tracker for that.
> 
> As for windows regression, it does not seem to be related to my patch.

Could you please create tickets for exposed issues?

**Comment by alexbatashev** [comment]
> > > @alexbatashev, the newly added test is failing on CUDA. Could you please fix it?
> > 
> > 
> > I disabled test on CUDA, because sycl tools seemingly do not generate enough information for the runtime to function correctly. I'll create a tracker for that.
> > As for windows regression, it does not seem to be related to my patch.
> 
> Could you please create tickets for exposed issues?

I added you to watchers for the created issues.

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
Could you split the test (move common part to the header)? You have just doubled test execution time which make cause timeouts on some configurations

**Comment by alexbatashev** [comment]
> Could you split the test (move common part to the header)? You have just doubled test execution time which make cause timeouts on some configurations

I switched to use of `if constexpr` which should save greatly on build times.

**Review by vladimirlaz — APPROVED** [review]

**Review by Pennycook — APPROVED** [review]

**Pull request merged**